### PR TITLE
feat(multitenant): refactor multitenant to be multi-app configurable

### PR DESCRIPTION
**Pull request by apeabody**
## 2-multitenant
- remove all hard coding for bank of anthos
- adds a new `apps` variable
```
variable "apps" {
  description = "Applications"
  type = map(object({
    ip_address_names = list(string)
  }))
}
```
- external ip addresses are now created based on `ip_address_names` in the new `apps` variable
- service accounts are now created for each app defined in the `apps` variable
- dynamic ip address test based on new global `AppNames = []string` testutils variable
- remove endpoints in favor of the updated dynamic ip address test
- dynamic service account test based on new global `AppNames = []string` testutils variable

### CODE REVIEW

File: 2-multitenant/modules/env_baseline/README.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ The following resources are created:
 
 | Name | Description | Type | Default | Required |
 |------|-------------|------|---------|:--------:|
+| apps | Applications | <pre>map(object({<br>    ip_address_names = list(string)<br>  }))</pre> | n/a | yes |
```
Comment (erictune) [inline 2-multitenant/modules/env_baseline/README.md:18]: Here or in a higher level README.md, define "app".

Some things to include in the definition:
- An "App" is high-level grouping of related services and workloads.
- That and App may include multiple namespaces, and also multiple team scopes.
- New apps are created infrequently. 
-  That Cymbal Bank example is one App, called `cymbal-bank`.
- That different Apps don't share IP addresses.
- A multi-tenant cluster can contain multiple apps.

File: test/integration/multitenant/multitenant_test.go [hunk 2]
```diff
@@ -118,23 +119,24 @@ func TestMultitenant(t *testing.T) {
...
-				assert.Subset(workloadIdentitiesSaListMembers, workloadIdentitiesUsers, fmt.Sprintf("service account %s should have workload identity users", workloadIdentitiesSAEmail))
-				assert.Equal(len(workloadIdentitiesUsers), len(workloadIdentitiesSaListMembers), fmt.Sprintf("service account % should have %d workload identity users", workloadIdentitiesSAEmail, len(workloadIdentitiesUsers)))
+				// App Service Accounts
+				for _, appName := range testutils.AppNames {
+					workloadIdentitiesSAEmail := fmt.Sprintf("%s@%s.iam.gserviceaccount.com", appName, clusterProjectID)
+					saOp := gcloud.Run(t, fmt.Sprintf("iam service-accounts describe %s --project %s", workloadIdentitiesSAEmail, clusterProjectID))
+					assert.False(saOp.Get("disabled").Bool(), "Service account should not be disabled.")
+
+					workloadIdentitiesUsers := []string{
+						fmt.Sprintf("serviceAccount:%s.svc.id.goog[accounts-%s/%s]", clusterProjectID, envName, appName),
```
Comment (erictune) [inline test/integration/multitenant/multitenant_test.go:129]: What is the point of using `appName` as the KSA name?  Would `default` be a better choice?

File: 2-multitenant/modules/env_baseline/wi.tf [hunk 3]
```diff
@@ -12,20 +12,25 @@
...
+  for_each = google_service_account.app_service_accounts
+
+  service_account_id = each.value.id
   role               = "roles/iam.workloadIdentityUser"
 
   members = [
-    "serviceAccount:${local.cluster_project_id}.svc.id.goog[accounts-${var.env}/bank-of-anthos]",
-    "serviceAccount:${local.cluster_project_id}.svc.id.goog[ledger-${var.env}/bank-of-anthos]",
+    "serviceAccount:${local.cluster_project_id}.svc.id.goog[accounts-${var.env}/${each.value.display_name}]",
+    "serviceAccount:${local.cluster_project_id}.svc.id.goog[ledger-${var.env}/${each.value.display_name}]",
```
Comment (erictune) [inline 2-multitenant/modules/env_baseline/wi.tf:33]: For a future PR: with BYOID, I had hoped we could delegate creation of KSAs to the app operator.  That means that this list is not part of a baseline, but instead different per namespace.

**Review by erictune — CHANGES_REQUESTED** [review]
Request adding documentation of what an App is.  Other comments are optional to address in this PR or later.

File: test/integration/multitenant/multitenant_test.go (hunk 2 above)
Comment (erictune) [reply test/integration/multitenant/multitenant_test.go:129]: This can be addressed later.

Comment (apeabody) [reply test/integration/multitenant/multitenant_test.go:129]: This was a conversion of the existing hard coded `bank-of-anthos` service account, however I'm not certain it's actually used/required: @yliaog?

My preference would be to drop from this stage entirely.

File: 2-multitenant/modules/env_baseline/README.md (hunk 1 above)
Comment (apeabody) [reply 2-multitenant/modules/env_baseline/README.md:18]: Sure - I'll add to the 2-multitenant README for this PR.  We'll need substantial changes to most of the other stages before the concept is "global".

**Review by erictune — APPROVED** [review]

**Pull request merged**